input/otlp/metadata.go

```diff
@@ -66,7 +66,8 @@ func translateResourceMetadata(resource pcommon.Resource, out *modelpb.APMEvent)
 			out.Service.Node.Name = truncate(v.Str())
 
 		// deployment.*
-		case semconv.AttributeDeploymentEnvironment:
+		// deployment.environment is deprecated, use deployment.environment.name instead
+		case semconv.AttributeDeploymentEnvironment, "deployment.environment.name":
```

Comment:
[to reviewer] No precedence between deprecated and new field, so the one appears later in the resource attributes wins.

Reply:
just a question on `semconv.AttributeDeploymentEnvironment`: is it possible for it to change in future releases of `semconv`? Or being deprecated applies some special handling?

Reply:
Good question. They won't change the constant, but will introduce a new one like https://github.com/open-telemetry/opentelemetry-collector/blob/semconv/v0.111.0/semconv/v1.27.0/generated_attribute_group.go#L1552

Reply:
BTW, we can also import semconv by versions to make things clear on what is latest and what is old - all semconvs are provided by the same module.

Something like `semconv18. AttributeDeploymentEnvironment` vs `semconv27.AttributeDeploymentEnvironmentName` - just an idea I have been playing around with for some other topics.